src/index.tsx

```diff
@@ -25,6 +25,55 @@ import { LiveChatLoaderProvider } from 'react-live-chat-loader';
 import { DialogManagerProvider } from 'app/contexts/dialog-manager/ActionDialogManager.context';
 import envService from 'app/core/services/env.service';
 
+const CANONICAL_DRIVE_HOSTNAME = 'drive.internxt.com';
+const CANONICAL_DRIVE_ORIGIN = `https://${CANONICAL_DRIVE_HOSTNAME}`;
+
+const buildSafeCanonicalUrl = () => {
+  const isBrowser = typeof globalThis !== 'undefined';
+
+  if (!isBrowser) {
+    return CANONICAL_DRIVE_ORIGIN;
+  }
+
+  try {
+    const { pathname, search, hash } = globalThis.location;
+    const relativeUrl = `${pathname}${search}${hash}`;
+    const normalizedUrl = new URL(relativeUrl, CANONICAL_DRIVE_ORIGIN);
+
+    if (normalizedUrl.origin !== CANONICAL_DRIVE_ORIGIN) {
+      return CANONICAL_DRIVE_ORIGIN;
+    }
+
+    return normalizedUrl.toString();
+  } catch {
+    return CANONICAL_DRIVE_ORIGIN;
+  }
+};
+
+const enforceCanonicalDriveDomain = () => {
+  const isBrowser = typeof globalThis !== 'undefined';
+
+  if (!isBrowser || !envService.isProduction()) {
+    return;
```

Comment:
Add tests for these functions. 
I believe this will break public shared links, as it will redirect to drive.internxt.com when entering another domain such as https://secure. internxt.com/sh/file/_gtOlt23Q5KBbyJqpMoT9w/eeae7a7d4b17278a85f600d3ff6b737cb18fbe96749841a5b695efcc1ef6aa28 for example.
Perhaps a whitelist of allowed domains should be added so that it does not redirect.